silabs-pti/src/main/java/com/silabs/pti/Main.java

```diff
@@ -311,4 +321,73 @@ private int runCommandSequence(final CommandLine cli) throws IOException {
     c.close();
     return 0;
   }
+
+  /**
+   * Convert arguments in properties file to CLI arguments format. 
+   * <br/><br/>
+   * NB: 
+   * 1) expected format is <code>-properties=path_to_properties_file</code>,
+   * where <code>path_to_properties_file</code> may be surrounded in double 
+   * quotes in case whitespace characters exists in path 
+   * 2) keys in properties file must be the same as CLI arguments, meaning they
+   * start with hyphen (e.g. <code>-ip</code> or <code>-delay</code>) 
+   * 3) in properties file, arguments without value will not have value 
+   * after '=' delimiter (e.g. <code>-discover=</code>).
+   * <br/><br/>
+   * Input Example (properties file content):
+   * <code><br/>
+   *  -zeroTimeThreshold=1000000<br/>
+      -format=text<br/>
+      -ip=1.2.3.4,9.8.7.6<br/>
+      -serial0=<br/>
+      -discreteNodeCapture=<br/>
+      -discover=<br/>
+   * </code>
+   *
+   * Output Example:
+   *  <code>-zeroTimeThreshold=1000000 -format=text -ip=1.2.3.4,9.8.7.6 -serial0 -discreteNodeCapture -discover</code>
+   *  <br/>
+   * @param progArgs array of args provided to silabs-pti.jar; <code>progArgs[0]</code> is
+   * argument <code>-properties=path_to_properties_file</code>
+   * @return array of strings with arguments in CLI format; empty array of strings
+   * if invalid input or error processing properties file; never null
+   */
+  private String[] convertPropsToArgs(String[] progArgs) {
+    if (progArgs.length == 1 && progArgs[0].startsWith(PROPERTIES)) {
```

Comment:
You get here because of condition on line 80:
```
if (args.length > 0 && args[0].startsWith(PROPERTIES)) {
```

So if args.length is, say 2, this whole method will do nothing.

I think we should allow people to say:
   -properties=file.properties -format=text

which would mean:
   1.) Take the CLI args from file.properties.
   2.) But then IGNORE the `format` in there, and use text regardless.

So i'd much more prefer the logic here, which would just work like this:

1.) If there is a `-properties=file.properties` argument, then run convertPropsToArgs.
2.) But then, STILL PARSE all the remaining args, and use those to override whatever was in -properties.

Reply:
OK. I'll update the logic based on your last suggesstion of 1) and 2): get arguments from `properties` file, then override them with values for other arguments provided at command line.